src/mount/mastercomm.cc

```diff
@@ -2118,6 +2121,13 @@ uint8_t fs_saureadchunk(std::vector<ChunkTypeWithAddress> &chunkservers, uint64_
 		setDisconnect(true);
 		return SAUNAFS_ERROR_IO;
 	}
+	for (auto& server : chunkservers) {
+		// If 127.0.0.1, let's assume it's the same as master
+		if (server.address.ip == localhost) {
+			server.address.ip=masterip;
+			safs::log_debug("changing chunkserver ip address 127.0.0.1 to {}", htonl(masterip));
+		}
+	}
```

Comment:
As a minor suggestion consider for readability to encapsulate this logic into a function for reusing it.

Reply:
Done